src/main/java/nl/nn/testtool/web/api/ReportApi.java

```diff
@@ -64,23 +64,26 @@
 	@Produces(MediaType.APPLICATION_JSON)
 	public Response getReport(@PathParam("storage") String storageParam,
 							  @PathParam("storageId") int storageId,
-							  @QueryParam("xml") @DefaultValue("false") boolean xml,
+							  @QueryParam("xml") @DefaultValue("false") boolean xml, // TODO: Remove this as its not needed
 							  @QueryParam("globalTransformer") @DefaultValue("false") boolean globalTransformer) {
 		try {
 			Storage storage = getBean(storageParam);
 			Report report = getReport(storage, storageId);
 			if (report == null)
 				return Response.status(Response.Status.NOT_FOUND).entity("Could not find report with id [" + storageId + "]").build();
 
+			String resultXml = report.toXml(); // Create an xml if it does not yet exist
+
+			// Old functionality: If the xml already existed, it would not carry out the transformation
 			if (globalTransformer) {
 				ReportXmlTransformer reportXmlTransformer = getBean("reportXmlTransformer");
-				if (reportXmlTransformer != null)
-					report.setGlobalReportXmlTransformer(reportXmlTransformer);
+				report.setGlobalReportXmlTransformer(reportXmlTransformer); // Set the transformer
+				resultXml = reportXmlTransformer.transform(resultXml); // If we have to transform, then transform
```

Comment:
Why don't you do the report.toXml() after the setGlobalReportXmlTransformer() so the report.toXml() will use the reportXmlTransformer and you don't need to do a manual reportXmlTransformer.transform(resultXml)?

Reply:
Because the report.toXml() will ignore the transformation if the XML already exists. I still have to add the flushed part functionality so the option you mentioned is a good solution, that's why I've converted the PR into the draft for now.